src/app/view/components/_common/site-selector/site-selector.component.ts

```diff
@@ -85,6 +86,19 @@ export class SiteSelectorComponent implements OnInit, OnChanges {
      */
     handleSitesRefresh(): void {
         this.paginationService.getCurrentPage().subscribe((items) => {
+            this.sites = items.splice(0);
+            this.getSitesList();
+
+            // if current site not being listed on updated sites list (socket message) then switch to default site
+            if (!this.isCurrentSiteOnListOfAvailableSites(this.sites, this.siteService.currentSite)) {
+                const defaultSite = this.getDefaultSite(this.sites);
+                this.siteChange({
+                    hostname: defaultSite.hostname,
+                    type: undefined,
```

Comment:
Why is this for?

Reply:
if the current site has been archived, then we change the current site in the dropdown for the default site

Reply:
why not just  do some like this: 'this.siteChange(defaultSite)'?

Reply:
I'll try this, but i'm not sure if what I mentioned on https://github.com/dotCMS/core-web/pull/576#discussion_r175796308 will happen...